common/tests/integration_tests.rs

```diff
@@ -128,12 +129,201 @@ fn basic2() {
+    csv.apply_commit(&Commit::Transaction(tx.clone())).unwrap();
+
+    let agenda = Agenda {
+        height,
+        author: rs.query_name(&author.0).unwrap(),
+        timestamp,
+        transactions_hash: Agenda::calculate_transactions_hash(&[tx.clone()]),
+    };
+    csv.apply_commit(&Commit::Agenda(agenda.clone())).unwrap();
+
+    csv.apply_commit(&Commit::AgendaProof(AgendaProof {
+        height,
+        agenda_hash: agenda.to_hash256(),
+        proof: keys[1..]
+            .iter()
+            .map(|(_, private_key)| TypedSignature::sign(&agenda, private_key).unwrap())
+            .collect::<Vec<_>>(),
+        timestamp,
+    }))
+    .unwrap();
+    let data = DelegationTransactionData {
+        delegator: rs.members[0].name.to_owned(),
+        delegatee: rs.members[2].name.to_owned(),
+        governance: true,
+        block_height: height,
+        timestamp,
+        chain_name: "pdao".to_owned(),
+    };
+    let tx_delegate = ExtraAgendaTransaction::Delegate(TxDelegate {
+        proof: TypedSignature::sign(&data, &keys[0].1).unwrap(),
+        data,
+    });
+    csv.apply_commit(&Commit::ExtraAgendaTransaction(tx_delegate))
+        .unwrap();
+
+    let block_header = BlockHeader {
+        author: author.0.clone(),
+        prev_block_finalization_proof: genesis_info.genesis_proof,
+        previous_hash: genesis_info.header.to_hash256(),
+        height,
+        timestamp,
+        commit_merkle_root: BlockHeader::calculate_commit_merkle_root(
+            &csv.get_total_commits()[1..],
+        ),
+        repository_merkle_root: Hash256::zero(),
+        validator_set: rs.get_validator_set().unwrap(),
+        version: genesis_info.header.version.clone(),
+    };
+    csv.apply_commit(&Commit::Block(block_header.clone()))
+        .unwrap();
+
+    let fp = keys
+        .iter()
+        .map(|(_, private_key)| TypedSignature::sign(&block_header, private_key).unwrap())
+        .collect::<Vec<_>>();
+    csv.verify_last_header_finalization(&fp).unwrap();
+    light_client
+        .update(block_header.clone(), fp.clone())
+        .unwrap();
+    let commits = csv.get_total_commits();
+    let merkle_tree = OneshotMerkleTree::create(
+        commits[1..=(commits.len() - 2)]
+            .iter()
+            .map(|c| c.to_hash256())
+            .collect(),
+    );
+    let merkle_proof = merkle_tree.create_merkle_proof(tx.to_hash256()).unwrap();
+    assert!(light_client.verify_transaction_commitment(&tx, height, merkle_proof));
+    assert_eq!(
+        csv.get_reserved_state()
+            .get_validator_set()
+            .unwrap()
+            .iter()
+            .find(|(pub_key, _)| pub_key == &keys[0].0),
+        None
+    );
+
+    let (height, timestamp) = (2, 1);
+    let mut csv =
+        CommitSequenceVerifier::new(block_header.clone(), csv.get_reserved_state().clone())
+            .unwrap();
+    let mut light_client = LightClient::new(block_header.clone());
+
+    let tx = Transaction {
```

Comment:
How about trying an empty agenda (without txes) for the second block? I think it's not important to add another tx for the next block.

Reply:
fixed